app/src/androidTest/java/com/github/h3lp3rs/h3lp/AwaitHelpActivityTest.kt

```diff
@@ -168,4 +178,46 @@ class AwaitHelpActivityTest {
 
         return intent
     }
-}*/
\ No newline at end of file
+
+    @Test
+    fun getsNotifiedWhenHelpIsComing() {
+        // Forge the right intent
+        val helpId = 1
+        val bundle = Bundle()
+        bundle.putInt(EXTRA_EMERGENCY_KEY, helpId)
+        bundle.putBoolean(EXTRA_CALLED_EMERGENCIES, true)
+        bundle.putStringArrayList(EXTRA_NEEDED_MEDICATION, arrayListOf(EPIPEN))
+        val intent = Intent(
+            getApplicationContext(),
+            AwaitHelpActivity::class.java
+        ).apply {
+            putExtras(bundle)
+        }
+        // Setup the database accordingly
+        val emergencyDb = MockDatabase()
+        val skills = HelperSkills(true, true, true, true,
+            true, true)
+        val emergency = EmergencyInformation(helpId.toString(), 2.0, 2.0, skills,
+            ArrayList(listOf("Epipen")), Date(), null, ArrayList())
+        emergencyDb.setObject(helpId.toString(), EmergencyInformation::class.java, emergency)
+        EMERGENCIES.db = emergencyDb
+        // Simulate arrival on await page after calling for help
+        launch<AwaitHelpActivity>(intent).use {
+            // Nobody coming
+            onView(withId(R.id.incomingHelpersNumber)).check(matches(withText("")))
+            // One person is coming
+            val helper1 = Helper(USER_TEST_ID + 1, 2.0, 2.0)
+            val withHelpers = emergency.copy(helpers = ArrayList(listOf(helper1)))
+            emergencyDb.setObject(helpId.toString(), EmergencyInformation::class.java, withHelpers)
+            onView(withId(R.id.incomingHelpersNumber)).check(matches(withText("1 person is coming to help you")))
+            // The same person is coming again, should NOT add a helper to the list
+            emergencyDb.setObject(helpId.toString(), EmergencyInformation::class.java, withHelpers)
+            onView(withId(R.id.incomingHelpersNumber)).check(matches(withText("1 person is coming to help you")))
+            // A second person is coming
+            val helper2 = Helper(USER_TEST_ID + 2, 2.1, 2.1)
+            val withMoreHelpers = emergency.copy(helpers = ArrayList(listOf(helper1, helper2)))
+            emergencyDb.setObject(helpId.toString(), EmergencyInformation::class.java, withMoreHelpers)
+            onView(withId(R.id.incomingHelpersNumber)).check(matches(withText("2 people are coming to help you")))
+        }
```

Comment:
I know, I did refactor part of this in the HelpPage, but since I only used this setup once it would have been a bit of a waste of time to do well.
The main issue is that creating a solid testing framework for all these situations would take 10h itself. Since I already went overtime this sprint, I didn't have the time to do it. Though I agree with your points...
Sadly, spending 10h on this won't add any new features :(

Reply:
Also, even if I put these values outside the function, it won't be any clearer. Like I said, the only real change would be to create an entire testing framework so that we can model complex scenarios. This is literally a full time job itself!